java/hive/src/main/java/io/v6d/hive/ql/io/VineyardOutputFormat.java

```diff
@@ -251,13 +319,14 @@ private static ArrowType toArrowType(TypeInfo typeInfo) {
                     case DOUBLE:
                         return Types.MinorType.FLOAT8.getType();
                     case STRING:
+                        return Types.MinorType.LARGEVARCHAR.getType();
                     case CHAR:
                     case VARCHAR:
-                        return Types.MinorType.LARGEVARCHAR.getType();
+                        return Types.MinorType.VARCHAR.getType();
```

Comment:
Use large var char for string types to avoid exceeding the size limit of arrow buffers in VarCharArray.

Reply:
Done